Install things where they should be and with the python bytecode instead
of into ${STAGEDIR}${WRKDIR}. Regressed by [1].

[1] https://code.qt.io/cgit/pyside/pyside-setup.git/commit/?h=6.6&id=90b3bab77a6465220051f52e72fe24bba862b606

--- PySide6/CMakeLists.txt.orig	2023-10-21 18:54:46 UTC
+++ PySide6/CMakeLists.txt
@@ -111,4 +111,4 @@ install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/QtAsync
         DESTINATION include/${BINDING_NAME}${pyside6_SUFFIX})
 
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/QtAsyncio"
-        DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
+        DESTINATION "${PYTHON_SITE_PACKAGES}/${BINDING_NAME}${pyside6_SUFFIX}")
